lib/rubocop/cop/rspec/return_from_stub.rb

```diff
@@ -40,20 +40,21 @@ class ReturnFromStub < Cop
         MSG_BLOCK = 'Use block for static values.'
 
         def_node_search :contains_stub?, '(send nil? :receive (...))'
+        def_node_matcher :stub_with_block?, '(block #contains_stub? ...)'
```

Comment:
It's confusing to call a node search from a node matcher, as in this case node search seems to work as a node matcher. It matches direct children, but not recursively.

Not sure why this is allowed. @bquorning @Darhazer do you know something about this?

Reply:
I didn't think of this usage, but it totally makes sense. The `#contains_stub?` is a method call that receives the node and returns boolean (it could be really a regular method inside the cop). So nothing prevents the method to perform a recursive search, no nothing prevents a node_search to be used inside the node_matcher.